### PR TITLE
📬 Add useNotificationsContext

### PR DESCRIPTION
**Pull request by nezouse**

### CODE REVIEW

**Comment by marekkirejczyk** [comment]
TODOL
- Add changeset (minor, for notifications and transactions)
- Solve conflicts

File: packages/core/src/hooks/useContractFunction.ts [hunk 1]
```diff
@@ -56,7 +57,7 @@ export function connectContractToSigner(contract: Contract, options?: Options, l
 
 export function useContractFunction(contract: Contract, functionName: string, options?: Options) {
   const [state, setState] = useState<TransactionStatus>({ status: 'None' })
-
+  const { addTransaction } = useContext(TransactionsContext)
```
Comment (marekkirejczyk) [inline packages/core/src/hooks/useContractFunction.ts:60]: Why not use:
`useTransactions()`

File: packages/core/src/providers/DAppProvider.tsx [hunk 2]
```diff
@@ -34,7 +36,11 @@ function DAppProviderWithConfig({ children }: WithConfigProps) {
         {readOnlyChainId && readOnlyUrls && (
           <ReadOnlyProviderActivator readOnlyChainId={readOnlyChainId} readOnlyUrls={readOnlyUrls} />
         )}
-        <ChainStateProvider multicallAddresses={multicallAddressesMerged}>{children}</ChainStateProvider>
+        <ChainStateProvider multicallAddresses={multicallAddressesMerged}>
+          <NotificationsProvider>
+            <TransactionProvider>{children}</TransactionProvider>
```
Comment (marekkirejczyk) [inline packages/core/src/providers/DAppProvider.tsx:41]: add newlines
```ts
<TransactionProvider>
  {children}
 </TransactionProvider>
```

File: packages/core/src/providers/notifications/model.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { TransactionReceipt, TransactionResponse } from '@ethersproject/abstract-provider'
+import { ChainId } from '../../constants'
+
+interface TransactionStarted {
+  type: 'transactionStarted'
+  transaction: TransactionResponse
+  submittedAt: number
+}
+
+interface TransactionSuccedd {
```
Comment (marekkirejczyk) [inline packages/core/src/providers/notifications/model.ts:10]: Succedd -> Succeed

File: packages/core/src/providers/notifications/model.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  type: 'transactionSuccedd'
```
Comment (marekkirejczyk) [inline packages/core/src/providers/notifications/model.ts:11]: Succeed

File: packages/core/src/hooks/useContractFunction.ts (hunk 1 above)
Comment (nezouse) [reply packages/core/src/hooks/useContractFunction.ts]: It's actually from a previous pull request, I forgot about these hooks, will change it in another pr.

File: packages/core/src/providers/DAppProvider.tsx (hunk 2 above)
Comment (nezouse) [reply packages/core/src/providers/DAppProvider.tsx:41]: I wish I could, but prettier prefers it this way.

File: packages/core/src/providers/transactions/provider.tsx [hunk 5]
```diff
@@ -46,6 +54,23 @@ export function TransactionProvider({ children }: Props) {
...
+                })
+              } else {
+                addNotification({
+                  type: 'transactionSucceed',
+                  submittedAt: Date.now(),
+                  transaction: tx.transaction,
+                  receipt,
+                  chainId,
+                })
+              }
```
Comment (marekkirejczyk) [inline packages/core/src/providers/transactions/provider.tsx:73]: Shorter:
```ts
const type = receipt.status === 0 ? 'transactionFailed' : 'transactionSucceed'
addNotification({
  type
  submittedAt: Date.now(),
  transaction: tx.transaction,
  receipt,
  chainId,
})
```

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**